frontend/packages/dev-console/src/components/project-access/ProjectAccess.tsx

```diff
@@ -23,21 +23,27 @@ import {
 } from './project-access-form-submit-utils';
 import { validationSchema } from './project-access-form-validation-utils';
 import ProjectAccessForm from './ProjectAccessForm';
-import { Verb, UserRoleBinding, Roles, roleBinding } from './project-access-form-utils-types';
+import { Verb, UserRoleBinding, roleBinding } from './project-access-form-utils-types';
 
 export interface ProjectAccessProps {
   formName: string;
   namespace: string;
   roleBindings?: { data: []; loaded: boolean; loadError: {} };
+  roles: Roles;
 }
 
-const ProjectAccess: React.FC<ProjectAccessProps> = ({ formName, namespace, roleBindings }) => {
+const ProjectAccess: React.FC<ProjectAccessProps> = ({
+  formName,
+  namespace,
+  roleBindings,
+  roles,
+}) => {
   const { t } = useTranslation();
-  if (!roleBindings.loaded && _.isEmpty(roleBindings.loadError)) {
+  if ((!roleBindings.loaded && _.isEmpty(roleBindings.loadError)) || _.isEmpty(roles)) {
     return <LoadingBox />;
   }
 
-  const filteredRoleBindings = filterRoleBindings(roleBindings.data, Roles);
+  const filteredRoleBindings = filterRoleBindings(roleBindings.data, roles);
```

Comment:
IMHO, the filter function must not need to know the frontend details, so we can just provide the keys here (the ClusterRole names)

```suggestion
  const filteredRoleBindings = filterRoleBindings(roleBindings.data, Object.keys(roles.data));
```